bob/src/cluster/operations.rs

```diff
@@ -19,13 +19,13 @@ pub(crate) async fn get_any(
 
 fn call_node_put(
     key: BobKey,
-    data: BobData,
+    data: Arc<BobData>,
     node: Node,
     options: PutOptions,
 ) -> JoinHandle<Result<NodeOutput<()>, NodeOutput<Error>>> {
     debug!("PUT[{}] put to {}", key, node.name());
     let task = async move {
-        LinkManager::call_node(&node, |conn| conn.put(key, data, options).boxed()).await
+        LinkManager::call_node(&node, |conn| conn.put(key, data.as_ref().clone(), options).boxed()).await
```

Comment:
Why the clone is on the current level? It is better to do it in `BobClient` level. Moreover, `data` field in `Blob` struct has `Vec<u8>` type, so, probably, the clone can be avoided

Reply:
Clone can't be avoided (since `BobClient` uses grpc which requires `vec`), and performing it on this level just seems more transparent (so it's obvious that `put` requires ownership).

Regarding moving clone into `BobClient` - you meand accept refs to Data? It will make this code clumsier since there will be arc clone and taking ref from it IMO.

Reply:
https://github.com/tokio-rs/prost/issues/31

Perhaps, converting type to `bytes::Bytes` will allow to avoid copying

Reply:
According to documentation on `Bytes` (https://docs.rs/bytes/latest/bytes/), this type is better suited for passing data inside a Bob than a vector. If `tonic` allows to use `Bytes` (look like it is: https://github.com/tokio-rs/prost/pull/341), we can achieve zero copying by receiving data in this type, storing it inside `BobData` in this type and transfer to remote nodes in this type without any conversion